spec/models/saved_claim_spec.rb

```diff
@@ -0,0 +1,227 @@
+          expect(StatsD).to have_received(:measure).with(
+            'saved_claim.time-to-file',
+            be_within(1.second).of(claim_duration),
+            tags: ["form_id:#{saved_claim.form_id}"]
+          )
+        end
+      end
+    end
+
+    context 'after destroy' do
+      it 'increments the saved_claim.destroy metric' do
+        saved_claim.save!
+        allow(StatsD).to receive(:increment)
+        saved_claim.destroy
+        expect(StatsD).to have_received(:increment).with('saved_claim.destroy',
+                                                         tags: ["form_id:#{saved_claim.form_id}"])
+      end
+    end
+  end
+
+  describe '#process_attachments!' do
+    let(:confirmation_code) { SecureRandom.uuid }
+    let(:form_data) { { some_key: [{ confirmationCode: confirmation_code }] }.to_json }
+
+    it 'processes attachments associated with the claim' do
+      attachment = create(:persistent_attachment, guid: confirmation_code, saved_claim: saved_claim)
+      saved_claim.process_attachments!
+
+      expect(attachment.reload.saved_claim_id).to eq(saved_claim.id)
+      expect(Lighthouse::SubmitBenefitsIntakeClaim).to have_enqueued_sidekiq_job(saved_claim.id)
+    end
+  end
+
+  describe '#confirmation_number' do
+    it 'returns the claim GUID' do
+      expect(saved_claim.confirmation_number).to eq(saved_claim.guid)
+    end
+  end
+
+  describe '#submitted_at' do
+    it 'returns the created_at' do
+      expect(saved_claim.submitted_at).to eq(saved_claim.created_at)
+    end
+  end
+
+  describe '#to_pdf' do
+    it 'calls PdfFill::Filler.fill_form' do
+      expect(PdfFill::Filler).to receive(:fill_form).with(saved_claim, nil)
+      saved_claim.to_pdf
+    end
+  end
+
+  describe '#update_form' do
+    it 'updates the form with new data' do
+      saved_claim.update_form('new_key', 'new_value')
+      expect(saved_claim.parsed_form['new_key']).to eq('new_value')
+    end
+  end
+
+  describe '#business_line' do
+    it 'returns empty string' do
+      expect(saved_claim.business_line).to eq('')
+    end
+  end
+
+  describe '#insert_notification' do
+    it 'creates a new ClaimVANotification record' do
+      saved_claim.save!
+
+      expect do
+        saved_claim.insert_notification(1)
+      end.to change(saved_claim.claim_va_notifications, :count).by(1)
+    end
+  end
+
+  describe '#va_notification?' do
+    let(:email_template_id) { 0 }
+
+    let!(:notification) do
+      ClaimVANotification.create(
+        saved_claim: saved_claim,
+        email_template_id:,
+        form_type: saved_claim.form_id,
+        email_sent: false
+      )
+    end
+
+    it 'returns the notification if it exists' do
+      expect(saved_claim.va_notification?(email_template_id)).to eq(notification)
+    end
+
+    it 'returns nil if the notification does not exist' do
+      expect(saved_claim.va_notification?('non_existent_template')).to be_nil
+    end
+  end
+end
```

Comment:
Please add newline to end of file

Reply:
It looks like it does? I think github may hide it in this view? Locally I see it in vs code unless I have some weird local config. I think this would have flagged it if this missing a new line?

Reply:
<img width="190" alt="Screenshot 2024-11-08 at 3 34 27 PM" src="https://github.com/user-attachments/assets/4d33fbca-6eb2-432d-b064-1d45afd53b4d">
¯\_(ツ)_/¯